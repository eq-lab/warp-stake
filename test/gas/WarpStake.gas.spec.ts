import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
import snapshotGasCost from '@uniswap/snapshot-gas-cost';
import { deployWarpStake } from '../shared/fixtures';
import { parseUnits } from 'ethers';
import { ethers } from 'hardhat';

describe('WarpStake gas', () => {
  it('deposit', async () => {
    const { warpToken, warpStake } = await loadFixture(deployWarpStake);
    const [_, user] = await ethers.getSigners();

    const input = parseUnits('1', await warpToken.decimals());
    await warpToken.connect(user).approve(warpStake.target, input);

    await snapshotGasCost(Number(await warpStake.connect(user).deposit.estimateGas(input)));
  });

  it('withdraw', async () => {
    const { transferManager, warpToken, warpStake } = await loadFixture(deployWarpStake);
    const [_, user] = await ethers.getSigners();

    const input = parseUnits('1', await warpToken.decimals());
    await warpToken.connect(user).approve(warpStake.target, input);
    await warpStake.connect(user).deposit(input);

    await warpStake.connect(transferManager).toggleWithdraws();

    await snapshotGasCost(Number(await warpStake.connect(user).withdraw.estimateGas()));
  });
});
