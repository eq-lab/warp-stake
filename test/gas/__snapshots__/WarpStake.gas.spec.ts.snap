// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarpStake gas deposit 1`] = `160960`;

exports[`WarpStake gas withdraw 1`] = `59986`;
